src/Validation/gen/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -164,4 +165,31 @@ internal static bool IsJsonIgnoredProperty(this IPropertySymbol property, INamed
             attr.AttributeClass is not null &&
             SymbolEqualityComparer.Default.Equals(attr.AttributeClass, jsonIgnoreAttributeSymbol));
     }
+
+    internal static bool IsSkippedValidationProperty(this IPropertySymbol property, INamedTypeSymbol skipValidationAttributeSymbol)
+    {
+        return property.HasAttribute(skipValidationAttributeSymbol) || property.Type.HasAttribute(skipValidationAttributeSymbol);
+    }
+
+    internal static bool IsSkippedValidationParameter(this IParameterSymbol parameter, INamedTypeSymbol skipValidationAttributeSymbol)
+    {
+        return parameter.HasAttribute(skipValidationAttributeSymbol) || parameter.Type.HasAttribute(skipValidationAttributeSymbol);
+    }
+
+    internal static bool HasAttribute(this ITypeSymbol? typeSymbol, INamedTypeSymbol attributeSymbol)
```

Comment:
Nit: Consider moving this to the RoslynUtils shared code so all the `HasAttribute` implementations are in one location.